sstables/sstable_set.cc

```diff
@@ -483,6 +490,12 @@ class min_position_reader_queue : public position_reader_queue {
             return {};
         }
 
+        if (_dummy_reader) {
```

Comment:
I wonder if we could internalize this logic to the `min_position_reader_queue()`. Create and return a dummy reader if we haven't return any readers yet and if a call to `pop(position_in_partition::before_all_clustered_rows{})` (meaning -- give me all you've got) and we have nothing to return.

Reply:
We could, although this would require `min_position_reader_queue` to take the partition key in the constructor which it needs to create the reader. So not 100% internal, it would kind of leak outside.

This would also restrict the behaviors of `min_position_reader_queue`, causing it to always return a non-empty reader. Maybe desirable, but maybe not as it also limits the extend of our tests (the tests which use `min_position_reader_queue` + `clustering_order_reader_merger` would not check how `clustering_order_reader_merge` behaves in all-readers-empty scenario).

Reply:
I don't see the problem with having to pass the partition key to the queue -- it operates on a single partition by definition, knowing its key feels natural, what doesn't feel natural is an optional dummy reader constructor parameter :D.

The fact that the bug cannot even happen by definition with `min_position_reader_queue` + `clustering_order_reader_merger` (with my proposal) sounds like a good thing to me.

I'm worried that with the current implementation, a corner case might go unnoticed (even in a future change) and we can have empty reads again. If we move the check to `pop()` we know all combination of inputs are covered, even all future ones.